lib/ansible/modules/cloud/amazon/emr_cluster.py

```diff
@@ -0,0 +1,458 @@
+  security_configuration:
+    description:
+      - name of security configuration
+  supported_products:
+    description:
+      - a list set by third-party when job is launched.
+  ec2_tags:
+    description:
+      - a dict with tags that will be associate with a cluster
+  visible_to_all_users:
+    description:
+      - define if cluster will be visible to all IAM users, default is True
+author:
+  - Wellington Moreira Ramos (@wmaramos)
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = '''
+- emr_cluster:
+    name: emr-jobs
+    region: us-east-1
+    log_uri: s3://bucket-name/logs/emr
+    wait: yes
+    ebs_optimized: yes
+    ebs_volumes:
+      - type: gp2
+        size: 300
+    configurations:
+      spark:
+        maximizeResourceAllocation: true
+      hbase-site:
+        hbase.rootdir: s3://bucket-name/hbase-site
+    applications:
+      - name: Hadoop
+      - name: Spark
+      - name: HBase
+    steps:
+      - name: s3-dist-cp
+        hadoop_jar_step:
+          jar: command-runner.jar
+          action_on_failure: TERMINATE_CLUSTER
+          args:
+            - "s3-dist-cp"
+            - "--src=s3://bucket-name/events"
+            - "--dest=hdfs://aggregate"
+            - "--groupBy=.*/[0-9]{4}/[0-9]{2}/.*gz"
+            - "--targetSize=128"
+            - "--outputCode=snappy"
+    instances:
+      ec2_key_name: ec2_key
+      ec2_subnet_id: subnet-923a31db
+      additional_slave_security_groups:
+        - sg-8eb2b1f1
+      additional_master_security_groups:
+        - sg-8eb2b1f1
+      instance_groups:
+        - name: master
+          instance_role: MASTER
+          instance_count: 1
+          instance_type: m3.xlarge
+          market: ON_DEMAND
+        - name: core
+          instance_role: CORE
+          instance_count: 4
+          instance_type: m3.xlarge
+          market: ON_DEMAND
+'''
+
+RETURN = '''
+# For more information see http://boto3.readthedocs.io/en/latest/reference/services/emr.html#EMR.Client.describe_cluster
+---
+cluster:
+  description: dictionary containing informations about EMR Cluster
+  returned: success
+  type: complex
+  contains:
+    id:
+      description: the unique id for the cluster
+      type: str
+      sample: j-2JNVMF38MZVHT
+    name:
+      description: the name of the cluster
+      type: str
+      sample: emr-jobs
+    status:
+      description: the current status defailts about the cluster
+      type: dict
+    ec2_instaces_attributes:
+      description: provides information about ec2
+      type: dict
+    log_uri:
+      description: path for logs location in S3
+      type: str
+    release_label:
+      description: the release label for the Amazon EMR
+      type: str
+    auto_terminate:
+      description: if true cluster should terminate after completing all step
+      type: boolean
+    applications:
+      description: list of the applications installed on this cluster
+      type: list
+    ec2_tags:
```

Comment:
This should also be returned as tags for consistency with other AWS modules. And it shouldn't be a list of `[{'Key', name1, 'Value', value1}, {'Key': name2, 'Value': value2},]`, but a dictionary like `{name1: value1, name2: value2}`. You can use the `boto3_tag_list_to_ansible_dict` function from ansible.module_utils.ec2 to make that transformation.

Reply:
done